src/js/productList.mjs

```diff
@@ -1,10 +1,9 @@
-import { getData} from "./productData.mjs";
-import { renderListWithTemplate} from "./utils.mjs";
-
+import { getData } from "./productData.mjs";
+import { renderListWithTemplate } from "./utils.mjs";
 
 function productCardTemplate(product) {
-    return `<li class="product-card">
-      <a href="product_pages/index.html?product=${product.Id}">
+  return `<li class="product-card">
+      <a href="/product_pages/index.html?product=${product.Id}">
```

Comment:
Make sure you pay attention to links. There was a `/` missing here which caused a 404 because it was using a relative path instead of an absolute path. See: https://www.youtube.com/watch?v=WbMVGiKnF3g